es-ds-components/components/es-modal.vue

```diff
@@ -0,0 +1,107 @@
+<script setup lang="ts">
+import PrimeDialog from "primevue/dialog";
+
+const props = defineProps({
+    hideFooter: {
+        type: Boolean,
+        default: false,
+    },
+    size: {
+        type: String,
+        default: "lg",
+    },
+});
+
+const modalPt = {
+    root: {
+        class: "modal-content",
+    },
+    mask: {
+        class: "p-dialog-mask p-component-overlay p-component-overlay-enter",
+    },
+    header: {
+        class: "modal-header",
+    },
+    title: {
+        class: "modal-title",
+    },
+    closeButton: {
+        class: "close",
+    },
+    content: {
+        class: "modal-body",
+    },
+    footer: {
+        class: "modal-footer",
+    },
+};
+
+const emit = defineEmits(["hidden"]);
+
+const onChange = (visible: boolean) => {
+    if (!visible) {
+        emit("hidden");
+    }
+};
+</script>
+
+<template>
+    <prime-dialog modal :class="`modal-${size}`" :pt="modalPt" @update:visible="onChange" dismissable-mask>
+        <template #header>
+            <h5 class="modal-title h2">
+                <slot name="modal-title" />
+            </h5>
+        </template>
+
+        <template #closeicon>
+            <icon-x class="text-gray-900" height="2rem" width="2rem" />
+        </template>
+
+        <slot />
+
+        <template #footer v-if="!hideFooter">
+            <slot name="modal-footer" />
+        </template>
+    </prime-dialog>
+</template>
+
+<style lang="scss">
```

Comment:
Could we use `scoped` here to avoid global CSS modifications? It'll be necessary to do `:deep()` on the selectors along with that to target the inner PrimeVue elements

Reply:
I ended up removing this code because it seemed like the wrong approach.